web3/main.py

```diff
@@ -538,6 +536,7 @@ def __init__(
                 "Provider must inherit from PersistentConnectionProvider class."
             )
         AsyncWeb3.__init__(self, provider, middlewares, modules, external_modules, ens)
+        self.ws = WebsocketConnection(self)
 
     # async for w3 in w3.persistent_websocket(provider)
     async def __aiter__(self) -> AsyncIterator["_PersistentConnectionWeb3"]:
```

Comment:
Since we have `Self` available in compat, it could be applied here and anywhere else a class returns an instance of itself 

```suggestion
    from web3._utils.compat import Self

    async def __aiter__(self) -> AsyncIterator[Self]:
```